src/main/java/org/truffleruby/language/dispatch/DispatchNode.java

```diff
@@ -127,7 +127,8 @@ public Object execute(VirtualFrame frame, Object receiver, String methodName, Ru
 
         final RubyClass metaclass = metaclassNode.execute(receiver);
 
-        if (isForeignCall.profile(metaclass == getContext().getCoreLibrary().truffleInteropForeignClass)) {
+        if (isForeignCall.profile(metaclass == getContext().getCoreLibrary().polyglotForeignObjectClass) &&
+                (LookupMethodNode.isForeignMethod(methodName))) {
             return callForeign(receiver, methodName, block, arguments);
```

Comment:
The logic here should change, I think we should `methodLookup.execute()` first, then if `method` is null/undefined, use a InteropNodes.InvokeMemberNode instead of `callMethodMissing` if it's a foreign object.

Then there is no need to hardcode any method name.

Reply:
My goal was to actually, if possible, remove all the methods in the hardcoded list and have them implemented in Ruby so that there is no need to check if the object if a polyglotForeignObjectClass.  

The hardcoded list has now only 10 methods.

I didn't want to push all the new chages in here before you took a look.  Should I push all the new changes to this pull request?

Reply:
Yeah, let's migrate them.

Reply:
Actually, if you do the change I suggest, there is no need to move these methods in this PR.
Methods defined in Ruby will be called, and anything else will just behave as before.

Methods that are defined in Kernel/BasicObject will be called with foreign objects, so those will need to be handle that.

Reply:
I changed the code around here, so now it's probably already the correct pattern if you rebase.
That is, use a Ruby method if there is one, and if there isn't use `callForeign()`.

You'll need to change `LookupMethodNode` so it doesn't just return `null` for foreign objects, but just continue with the regular lookup.